web-app/src/lib/services/agent.service.ts

```diff
@@ -45,6 +45,13 @@ export const agentService = (() => {
     agent: AgentWithOrganizations,
     userOrganizationIds: string[],
   ): boolean {
+    // Blacklist: deny if user belongs to any organization that blacklisted this agent
+    const isBlacklisted = agent.blacklistedOrganizations.some((org) =>
+      userOrganizationIds.includes(org.id),
+    );
+    if (isBlacklisted) return false;
```

Comment:
Isn't it wrong? I believe it should only be blacklisted if the active scope involves the blacklisted organization. Why should my personal account be affected by an organization's decision?

Reply:
I saw that, too, but as we didn't had access to the session here before, I thought was ok. But I agree with you that my personal profile shouldn't be affect, I'll add the session here and check if that's the active scope.